src/odemis/driver/simcam.py

```diff
@@ -142,6 +143,48 @@ def __init__(self, name, role, image, dependencies=None, daemon=None, blur_facto
         # Convenience event for the user to connect and fire
         self.softwareTrigger = model.Event()
 
+        # show_message(None, 'starting and error wise thread', 'test error', timeout = 1.0)
+        self._errorCreationThread = threading.Thread(target = self._state_error_creation,
+                                                     args = (),
+                                                     name = "Creating and state error")
+        self._errorCreationThread.daemon = True
+        self._errorCreationThread.start()
+
+    def _state_error_creation(self):
+        '''
+        by creating the files 'simcam.HW.error' or 'simcam.HW.good' in the folder '/var/run/odemisd' and hardware
+        error can be produced in the Sim Cam or the error can be 'fixed'
+        After the error is produced the files should be deleted
+        :return:
+        '''
+        def attempt_to_delete_file(del_file):
+            try:
+                os.remove(del_file)
+            except:
+                pass
+
+        path = '/var/run/odemisd'
+        [attempt_to_delete_file(del_file) for del_file in [path + '/simcam.HW.error', path + '/simcam.HW.good']]
+
+        tnext = 0
+        while True:
+            tnow = time.time()
+
+            # sleep a bit to avoid refreshing too fast
+            tsleep = tnext - tnow
+            if tsleep > 0.001:
+                time.sleep(tsleep)
+            tnext = time.time() + 0.1  # max 10 Hz
```

Comment:
No need to make it so complex, as the rest of the block has a (approximately) fixed duration.
=> Replace by `time.sleep(0.1) # max 10 Hz`.